File: Calculate_age.py
import time
from calendar import isleap

def calculate_age(name, birth_year, birth_month, birth_day):
    current_time = time.localtime(time.time())
    current_year = current_time.tm_year
    current_month = current_time.tm_mon
    current_day = current_time.tm_mday

    leap_year = isleap(current_year)

    if birth_year > current_year:
        print("Invalid birth year. It's in the future.")
        return


    if birth_month < 1 or birth_month > 12:
        print("Invalid birth month. It should be between 1 and 12.")
        return

    if birth_day < 1 or birth_day > 31:
        print("Invalid birth day. It should be between 1 and 31.")
        return


    age_years = current_year - birth_year

    age_months = 0
    
    age_days = 0
   
    if(current_month < birth_month or current_month==birth_month and current_day<birth_day):
        ageYears-=1

    if(current_month>birth_month):
        age_months=current_month-birth_month
    else:
        age_months=12-(birth_month-current_month)
    age_days+=month_days(birth_month,leap_year)-birth_day
    for i in range(birth_month+1,13):
        age_days+=month_days(i,leap_year)    
    
    for year in range(birth_year+1, current_year):
        if not isleap(year):

            age_days += 365
        else:
            age_days += 366

    for month in range(1, current_month):
        age_days += month_days(month, leap_year)

    age_days += current_day

    print(f"{name}'s age is {age_years} years, {age_months} months, or {age_days} days.")

def month_days(month, leap_year):
    if month in [1, 3, 5, 7, 8, 10, 12]:
        return 31
    elif month in [4, 6, 9, 11]:
        return 30
    elif month == 2:
        return 29 if leap_year else 28

if __name__ == "__main__":
    name = input("Input your name: ")
    birth_year = int(input("Input your birth year: "))
    birth_month = int(input("Input your birth month: "))
    birth_day = int(input("Input your birth day: "))

    calculate_age(name, birth_year, birth_month, birth_day)

File: Calculate_age.java
import java.util.Calendar;

public class Calculate_age {

    public static void calculateAge(String name, int birthYear, int birthMonth, int birthDay) {
        Calendar currentTime = Calendar.getInstance();
        int currentYear = currentTime.get(Calendar.YEAR);
        int currentMonth = currentTime.get(Calendar.MONTH) + 1; 
        int currentDay = currentTime.get(Calendar.DAY_OF_MONTH);
        boolean leapYear = isLeap(currentYear);

        int ageYears = currentYear - birthYear;

        int ageMonths = (currentYear - birthYear) * 12 + currentMonth - birthMonth;

        int ageDays = (currentYear - birthYear) * 365;

        for (int year = birthYear; year < currentYear; year++) {
            if (isLeap(year)) {
                ageDays += 365;
            } else {
                ageDays += 366; 
            }
        }

        for (int month = birthMonth; month < currentMonth; month++) {
            ageDays += monthDays(month, leapYear);
        }
        
        ageDays += currentDay - birthDay;

        System.out.println(name + "'s age is " + ageYears + " years, " + ageMonths + " months, or " + ageDays + " days.");
    }

    public static int monthDays(int month, boolean leapYear) {
        if (month == 2) {
            return leapYear ? 29 : 28;
        } else if (month == 4 || month == 6 || month == 9 || month == 11) {
            return 30;
        } else {
            return 31;
        }
    }

    public static boolean isLeap(int year) {
        return year % 4 == 0 && (year % 100 != 0 || year % 400 == 0);
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Input your name: ");
        String name = scanner.nextLine();
        System.out.print("Input your birth year: ");
        int birthYear = scanner.nextInt();
        System.out.print("Input your birth month: ");
        int birthMonth = scanner.nextInt();
        System.out.print("Input your birth day: ");
        int birthDay = scanner.nextInt();

    // public static void calculateAge(name, birthYear, birthMonth, birthDay);
    public static void calculateAge(String name, int birthYear, int birthMonth, int birthDay) {
        // Create a Calendar object to represent the current time.
        Calendar currentTime = Calendar.getInstance();
    
        // Get the current year, month, and day of month.
        int currentYear = currentTime.get(Calendar.YEAR);
        int currentMonth = currentTime.get(Calendar.MONTH) + 1;
        int currentDay = currentTime.get(Calendar.DAY_OF_MONTH);
    
        // Check if the current year is a leap year.
        boolean leapYear = isLeap(currentYear);
    
        // Calculate the age in years, months, and days.
    
        
    int ageYears = currentYear - birthYear;
        int ageMonths = (currentYear - birthYear) * 12 + currentMonth - birthMonth;
        int ageDays = (currentYear - birthYear) * 365;
    
        // Add the number of days in each year between the birth year and the current year,
    
        
      // taking into account leap years.
    
        
    for (int year = birthYear; year < currentYear; year++) {
        if (isLeap(year)) {
            ageDays += 366;
        } else {
            ageDays += 365;
        }
    
        // Add the number of days in each month between the birth month and the current month,
    
        
      // taking into account leap years.
    
        
    for (int month = birthMonth; month < currentMonth; month++) {
        ageDays += monthDays(month, leapYear);
        }
    
        // Add the number of days in the current month.
        ageDays += currentDay - birthDay;
    
        // Print the age in years, months, and days.
        System.out.println(name + "'s age is " + ageYears + " years, " + ageMonths + " months, or " + ageDays + " days.");
    }
    
    


    }
}
